source/_posts/2016-09-14-what_the_tag_series_ep1_document_object.markdown

```diff
@@ -0,0 +1,123 @@
+
+<!--more-->
+
+I've been working with HTML for 2 years, but I am pretty sure I am going to learn as much as many people with this adventure, so don't worry, if you just started coding and you feel that your sites still look like [the world's worst website ever](http://www.theworldsworstwebsiteever.com/) we will learn, and that's the important thing.
+
+# The Document object
+
+The DOM (Document object model) standard defines a [Document](https://dom.spec.whatwg.org/#interface-document) interface which has many attributes and functions that will let you interact and know more about the document you are dealing with.
+
+**Please right click this website, click inspect, go to the console tab and try the things we learn as you read, you won't be breaking anything!**
+
+## Resource metadata management
+
+`document.referrer`
+
+This attribute returns the URL of the document from which the user navigated from or the empty string if there is no such information (for example if you typed the URL directly) or if this information was blocked using the `noreferrer` link type.
+
+`document.cookie`
+
+Very common these days, if you are not using it in your websites probably someone is using it on you. It is used to store the HTTP cookies for the current document, this can be set by you or it can be set automatically by the browser. In some cases it will throw an error, see the [definition of the cookie](https://html.spec.whatwg.org/multipage/dom.html#dom-document-cookie) for more info.
+
+`document.lastModified`
+
+It lets you know about when the document was modified for the last time, this info is retrieved from the server response, in case this is not available the current time is returned in the local time zone.
+
+`document.readyState`
+
+This readonly attribute is a very important and useful one, it holds the current state of the document, it can be in three different states:
+
+**loading**
+When the document has not finished loading (the response from the server is not yet completed).
+
+**interactive**
+In case the first load is done and the parsing process is completed but there are some resources that need to be loaded, for example images, styles, javascript, etc.
+
+**complete**
+When all related resources are loaded. Once this state is achieved the `load` event will be fired.
+
+Every time the value of this attribute changes the `readystatechange` event is fired. This event is equivalent to the more known ones `load` and `DOMContentLoaded`, next is an example, that I borrowed from the docs of [Mozilla](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState) (if you need more cool examples just visit them!), this simply listens to the event and does some application initialization:
+
```

Comment:
I would probably separate the sentence. `This simply listens to the...`, either keep it here or below the example (I thinks it makes more sense below, but it might not look well)